src/component/main/Editor.js

```diff
@@ -0,0 +1,79 @@
+import request from "../../api.js";
+
+export default class Editor {
+
+    constructor({ pageElement, onChangeTitle }) {
+        this.onChangeTitle = onChangeTitle.bind(this);
+        const boardElement = document.createElement('div');
+        this.titleEditorElement = document.createElement('h1');
+        this.editorElement = document.createElement('form');
+
+        boardElement.className = "editor";
+        this.editorElement.className = "textEditor";
+        this.titleEditorElement.placeholder = "제목 없음";
+        this.titleEditorElement.setAttribute("contenteditable", "true");
+        this.editorElement.setAttribute("contenteditable", "true");
+
+        pageElement.appendChild(boardElement);
+        boardElement.appendChild(this.titleEditorElement);
+        boardElement.appendChild(this.editorElement);
+
+        this.setEvent();
+    }
+
+    convertHeadingTag(textHTML, e) {
+        const findH1 = textHTML.split("#&nbsp;");
+        const findH2 = textHTML.split("##&nbsp;");
+        const findH3 = textHTML.split("###&nbsp;");
+        const findH4 = textHTML.split("####&nbsp;");
+        const findHeadingTagArr = [findH1, findH2, findH3, findH4];
+
+        findHeadingTagArr.map((item, index) => {
+            if (item.length > 1) {
+                e.target.innerHTML = item[0] + `<h${index + 1} id="cur">&nbsp;</h${index + 1}>`;
+                const selection = window.getSelection();
+                const range = document.createRange();
+                const HeadingTag = document.getElementById('cur');
+                range.selectNode(HeadingTag);
+                HeadingTag.id = "";
+                selection.removeAllRanges();
+                selection.addRange(range);
+            }
+        });
+    }
+
+    async updateDocument(title, content) {
+        await request(`/documents/${this.id}`, {
+            method: `PUT`,
+            body: JSON.stringify({
+                "title": title,
+                "content": content
+            })
+        });
+        if (this.title !== title) {
+            this.title = title;
+            this.onChangeTitle(this.id, title);
+        }
+    }
+
+    setEvent() {
+        this.titleEditorElement.addEventListener("keyup", (e) => {
+            const titleTextContent = e.target.textContent.length > 0 ? e.target.textContent : "제목 없음";
+            this.updateDocument(titleTextContent, this.editorElement.textHTML);
+        });
+
+        this.editorElement.addEventListener("keyup", (e) => {
+            const textHTML = e.target.innerHTML;
+            const findDiv = this.editorElement.querySelector('div');
+
+            if (findDiv) {
+                e.preventDefault();
+                const newLine = document.createElement('br');
+                this.editorElement.removeChild(findDiv);
+                this.editorElement.appendChild(newLine);
+            }
```

Comment:
저는 if문으로 인한 중첩이 코드 가독성을 떨어뜨린다는 말을 들은 이후로 최대한 중첩을 피하려고 노력하고 있습니다! 그 방법 중 하나가 early return이에요! `findDiv`가 없다면 return 해버리면 중첩을 피할 수 있을 것 같아요
```suggestion
              if (!findDiv) return;
              e.preventDefault();
              const newLine = document.createElement('br');
              this.editorElement.removeChild(findDiv);
              this.editorElement.appendChild(newLine);
           
```

Reply:
저도 가독성 부분 때문에  early return사용을 지향하고있는데, 이번 부분은 어째서인지 빠트린것 같네요. 하하

Reply:
```
 this.editorElement.addEventListener("keyup", (e) => {
            const textHTML = e.target.innerHTML;
            const findDiv = this.editorElement.querySelector('div');
            if (findDiv) {
                e.preventDefault();
                const newLine = document.createElement('br');
                this.editorElement.removeChild(findDiv);
                this.editorElement.appendChild(newLine);
            }
            this.convertHeadingTag(textHTML, e);
            this.updateDocument(this.titleEditorElement.textContent, textHTML);
        });
```
오호 저도 가독성이유로 ealry return을 지향하는 편입니다.
 하지만 해당 이벤트 함수는 findDiv 조건문 블록만 사용되는게 아닌 76~77번째 줄이 필수적으로 사용해야합니다. 정확히는 findDIv 조건문 블록은 76 ~77 번째 줄의 함수가 사용할 수 있도록 돔에 div 블록으로 생성된게 있다면 \<br\>로 바꿔주는 역할 만 하고 있죠.
 아마 제가 이 관련 설명을 주석으로 달지 않거나  findDiv->br 변환 부분을 함수로 분리하지 않아 생긴 이슈인거같습니다.